src/Components/SysDetailsTable/NewSysDetailsTable.js

```diff
@@ -91,30 +93,48 @@ const NewSysDetailsTable = ({ systemId }) => {
     ...initialState,
   });
 
+  const handleSetSearchParams = (ruleName, status) => {
```

Comment:
Here, we can also apply filtering out the empty params and then set valid params only once. As @bastilian suggested one hook would make this more robust&clean